src/main/java/htsjdk/variant/variantcontext/VariantContextBuilder.java

```diff
@@ -342,23 +356,52 @@ private void makeAttributesModifiable() {
      * @return this builder
      */
     public VariantContextBuilder filters(final Set<String> filters) {
-        this.filters = filters;
+        makeFiltersModifiable();
+        if (filters == null) {
+            unfiltered();
+        } else {
+            this.filters.clear();
+            this.filters.addAll(filters);
+            toValidate.add(VariantContext.Validation.FILTERS);
+        }
         return this;
     }
 
+    /**
+     * This builder's filters are set to this value
+     *
+     * filters can be <code>null</code> -&gt; meaning there are no filters
+     *
+     * @param filters Set of strings to set as the filters for this builder
+     * @return this builder
+     */
+    private void filtersAsIs(final Set<String> filters) {
+        this.filters = filters;
+        toValidate.add(VariantContext.Validation.FILTERS);
+    }
+
+
     /**
      * {@link #filters}
      *
      * @param filters  Set of strings to set as the filters for this builder
      * @return this builder
      */
     public VariantContextBuilder filters(final String ... filters) {
-        filters(new LinkedHashSet<String>(Arrays.asList(filters)));
+        filtersAsIs(new LinkedHashSet<>(Arrays.asList(filters)));
```

Comment:
How is this different than `filters(new LinkedHashSet<>(Arrays.asList(filters)))`?

Reply:
doesn't bother call make filters modifyable.